src/wattsi.pas

```diff
@@ -1036,6 +1036,37 @@    TCrossReferences = record
          end
       end
       else
+      if (HasAncestor(Element, nsHTML, eTD) or HasAncestor(Element, nsHTML, eDT)) then
+      begin
+         // For elements we're annotating inside a <td> or <dt>, we append the
+         // annotation directly to the <td> or <dt> as the last child. Otherwise
+         // in cases where we have a long table (e.g., the list of events in the
+         // index) or long <dl> list (e.g., the list of pseudo-classes), all the
+         // annos for everything in that table or list end up being merged into
+         // a single annotation way up at the beginning of the table or list.
+         ClassName := 'mdn-anno wrapped before';
+         Candidate := Element;
+         repeat
+            Candidate := Candidate.ParentNode;
+            if (not (Candidate is TElement)) then exit;
+         until (TElement(Candidate).IsIdentity(nsHTML, eTD) or
+                TElement(Candidate).IsIdentity(nsHTML, eDT));
+         TargetAncestor := TElement(Candidate);
+         if ((TargetAncestor.LastChild is TElement) and
```

Comment:
This might benefit from a LastChildElement helper, since whitespace as the last child should not prevent this consolidation. Not sure if it makes a difference in practice, though.

Reply:
> This might benefit from a LastChildElement helper, since whitespace as the last child should not prevent this consolidation. Not sure if it makes a difference in practice, though.

Not sure if it does either, but I went ahead and added it :)